### PR TITLE
Adds fontScalePresets

### PR DESCRIPTION
**Pull request by frytyler**
This adds 5 presets from [ModularScale.com](http://www.modularscale.com/) that can be used by passing a string reference through `baseFontScale`. If the value passed to `baseFontScale` does not match a preset an error will be thrown. 

Previous behaviour of passing a custom array is still support and still restricts to length of 6. 

I've updated the `README.md` to best describe this feature but feel free to make some suggestions there. 

**Note** I had to disable one line in the test for eslint as it didn't like a `new` call without a variable assignment however a variable assignment also was against the eslint rules. Thoughts?

Anyway appreciate your input on the issue on direction, let me know if you need anything else.

Completes #2

### CODE REVIEW

File: src/utils/index.js [hunk 1]
```diff
@@ -16,6 +18,18 @@ export const trimArrayToMaxOf6 = array => {
...
+export const getFontScale = fontScale => {
+  if (Array.isArray(fontScale)) {
+    return trimArrayToMaxOf6(fontScale)
+  }
+
+  if (fontScalePresets.hasOwnProperty(fontScale)) {
+    return fontScalePresets[fontScale]
+  } else {
+    throw new Error(`No Font Scale Preset Found for "${fontScale}", the presets available are: "${Object.keys(fontScalePresets)}"`)
+  }
```
Comment (kyleshevlin) [inline src/utils/index.js]: I really like this. Especially informing the user in the error of the available presets.

File: src/index.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ export default class Shevy {
     this.baseSpacing = this.baseSpacing.bind(this)
 
     // Set headings
-    baseFontScale.forEach((factor, index) => {
+    this.baseFontScale.forEach((factor, index) => {
```
Comment (kyleshevlin) [inline src/index.js]: Good catch.

File: src/constants/index.js [hunk 3]
```diff
@@ -1,3 +1,11 @@
+export const fontScalePresets = {
+  major_second: [1.802, 1.602, 1.424, 1.266, 1.125, 1],
+  minor_third: [2.488, 2.074, 1.728, 1.44, 1.2, 1],
+  major_third: [3.052, 2.441, 1.953, 1.563, 1.25, 1],
+  perfect_fourth: [4.209, 3.157, 2.369, 1.777, 1.333, 1],
+  augmented_fourth: [5.653, 3.998, 2.827, 1.999, 1.414, 1]
```
Comment (kyleshevlin) [inline src/constants/index.js]: Could you update these to be camelCased? I prefer that in JavaScript. Might even add a ESLint rule for it.

File: __tests__/utils/index.test.js [hunk 4]
```diff
@@ -71,6 +72,30 @@ describe('Utils', () => {
     })
   })
 
+  describe('getFontScale', () => {
+    const { getFontScale } = utils
+
+    describe('baseFontScale as an array', () => {
+      const baseFontScaleAsArray = [1, 2, 3, 4, 5, 6]
+
+      it('returns array with modification', () => {
```
Comment (kyleshevlin) [inline __tests__/utils/index.test.js]: Think you meant `without`

File: __tests__/index.test.js [hunk 5]
```diff
@@ -357,28 +358,52 @@ describe('Shevy', () => {
...
         })
 
-        it('should trim array to length of 6, removing from the end', () => {
-          const expectedArray = [7, 6, 5, 4, 3, 2]
-          expect(shevy.baseFontScale).toEqual(expectedArray)
+        describe('has an invalid preset name', () => {
+          it('should throw an error if preset does not exist', () => {
+            expect(() => {
+              // eslint-disable-next-line
+              new Shevy({ baseFontScale: 'not_valid' })
```
Comment (kyleshevlin) [inline __tests__/index.test.js]: I think this is fine for now. You can't set a variable name and not use it, so I understand why that is failing. Perhaps get the specific error and disable it. I think that would be `// eslint-disable-next-line no-new`

File: README.md [hunk 6]
```diff
@@ -78,7 +78,21 @@ This is used to determine the line height calculations in Shevy. Each line heigh
...
+Font Scale Presets are available based [ModularScale.com](http://www.modularscale.com/). If `baseFontScale` is a string it will be used to match by key to the following presets:
+
+```javascript
+{
+  major_second: [1.802, 1.602, 1.424, 1.266, 1.125, 1],
+  minor_third: [2.488, 2.074, 1.728, 1.44, 1.2, 1],
+  major_third: [3.052, 2.441, 1.953, 1.563, 1.25, 1],
+  perfect_fourth: [4.209, 3.157, 2.369, 1.777, 1.333, 1],
+  augmented_fourth: [5.653, 3.998, 2.827, 1.999, 1.414, 1]
+}
```
Comment (kyleshevlin) [inline README.md]: Update these to camelCase please.

**Comment by kyleshevlin** [comment]
@frytyler This is excellent. I have made some comments on the work that I would like to see changed. Nothing major. Thank you so much for this. Let's get this merged soon!

File: src/constants/index.js (hunk 3 above)
Comment (frytyler) [reply src/constants/index.js]: Ya great call, I was blinding naming as the modularscale pattern was but you are right camelCase for the win

File: src/index.js (hunk 2 above)
Comment (frytyler) [reply src/index.js]: Test for the win here. Failed one and was able to fine this.

**Comment by frytyler** [comment]
@kyleshevlin from my initial push there are new `dist` files. I won't commit them in the next push however would you like them pushed to the branch or do you have something to generate them when you merge?

**Comment by kyleshevlin** [comment]
@frytyler if you used the commitizen script to make your commits, the build should have been run automatically for you (so that commits keep what would be the distributed code up to date). Did you somehow remove the dist files post-commit, pre-push?

**Comment by frytyler** [comment]
I did use commitizen, I never removed them. I should clarify they aren't new distributed files they are just the modified ones. So I assume that means you want me to go ahead and push that. I'll do that now.

**Comment by kyleshevlin** [comment]
Thanks for all the work @frytyler, this looks great!

**Pull request merged**